src/utils/keys-utils.ts

```diff
@@ -157,16 +159,14 @@ async function getSharedSymmetricKey(
   privateKey: Uint8Array,
   ephemeralKey: Uint8Array,
 ): Promise<Optional<Uint8Array>> {
+  if (typeof initCurve25519wasm === 'function') await initCurve25519wasm();
```

Comment:
Should we init at the start of this file, rather than on every `getCommitmentBlindingKey` call?

Reply:
Where in the start of the file? I'm under the impression we can't use await on the top-level. Can we?

Reply:
Try like this: 

```
(async () => {
  if (typeof initCurve25519wasm === 'function') await initCurve25519wasm();
})();
```

Reply:
This creates an anonymous function and calls it immediately. Can be used at the top of this file.

Reply:
Ah, sure, I can do that, but it could theoretically have a race condition, like the getSharedPrivateKey (or what's it called again) function isn't called right at the start of the program before the init has had chance to complete. You know this codebase better than me, is there some kind of "front door" API where things get initialized? That would be the best place for the init.

Reply:
Actually, why don't we leave it as is. I don't like race conditions. 

If the curve is already init, will it simply return?

Reply:
> If the curve is already init, will it simply return?

Good point. Actually, I think it won't. So I'll call the `init()` outside, but preserve the promise output, so we can just `await` the promise output to guarantee that we don't double-init.

Reply:
Great plan.